user.go

```diff
@@ -37,53 +110,87 @@ func (h UserHandler) AddUserHandler(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
+	
+	user.ID = primitive.NewObjectID()
 
-	response, err := json.Marshal(user)
+	insertResult, err := h.Collection.InsertOne(context.Background(), user)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	
+	insertResponse, err := json.Marshal(insertResult.InsertedID)
 
 	w.Header().Set("Content-Type", "application/json")
-	w.Write(response)
+	w.Write(insertResponse)
 }
 
 // UpdateUserHandler Gets a PUT request with user's data in the request's body
```

Comment:
is the function get a complete user data or it can be partial?

Reply:
It can be partial, I didn't implement validation for the user info to be complete.